nilearn/surface/surface.py

```diff
@@ -1014,3 +1014,235 @@ def check_surface(surface):
+        The surface whose adjacency matrix is to be computed.
+
+    values : { 'len' | 'invlen' | 'ones'}, optional
+        If `values` is `'ones'` (the default), then the returned matrix
+        contains uniform values in the cells representing edges. If the value is
+        `'len'` then the cells contain the edge length of the represented
+        edge. If the value is `'invlen'`, then the the inverse of the distances
+        are returned.
+
+    dtype : numpy dtype-like or None, optional
+        The dtype that should be used for the returned sparse matrix.
+
+    Returns
+    -------
+    matrix : scipy.sparse.csr_matrix
+        A sparse matrix representing the edge relationships in `surface`.
+
+    """
+    from scipy.sparse import csr_matrix
+    surface = load_surf_mesh(surface)
+    # This is a bit of a hack to quickly find a unique set of all edges.
+    n = surface.coordinates.shape[0]
+    edges = np.vstack([surface.faces[:,[0,1]],
+                       surface.faces[:,[0,2]],
+                       surface.faces[:,[1,2]]])
+    edges = edges.astype(np.int64)
+    bigcol = edges[:,0] > edges[:,1]
+    lilcol = ~bigcol
+    edges = np.concatenate([edges[bigcol,0] + edges[bigcol,1]*n,
+                            edges[lilcol,1] + edges[lilcol,0]*n])
+    edges = np.unique(edges)
+    (u,v) = (edges // n, edges % n)
+    # Calculate distances between pairs. We use this as a weighting to make sure that
+    # smoothing takes into account the distance between each vertex neighbor
+    if values == 'len' or values == 'invlen':
+        coords = surface.coordinates
+        edge_lens = np.sqrt(np.sum((coords[u,:] - coords[v,:])**2, axis=1))
+        if dtype is None:
+            dtype = edge_lens.dtype
+        else:
+            edge_lens = edge_lens.astype(dtype)
+        if values == 'invlen':
+            edge_lens = 1 / edge_lens
+    elif values == 'ones':
+        if dtype is None:
+            edge_lens = np.ones_like(edges)
+        else:
+            edge_lens = np.ones(edges.shape, dtype=dtype)
+    else:
+        raise ValueError(f"unrecognized values argument: {values}")
+    # We can now make a sparse matrix.
+    ee = np.concatenate([edge_lens, edge_lens])
+    uv = np.concatenate([u, v])
+    vu = np.concatenate([v, u])
+    return csr_matrix((ee, (uv, vu)), shape=(n,n))
+
+def compute_vertex_neighborhoods(surface):
+    """For each vertex, compute the neighborhood.
+    The neighborhood is  defined as all the vertices that are connected by a
+    face.
+    
+    Parameters
+    ----------
+    surface : Surface-like
+        The surface whose vertex neighborhoods are to be computed.
+    
+    Returns
+    -------
+    neighbors : list
+        A list of all the vertices that are connected to each vertex
+    """
+    from scipy.sparse import find
+    matrix = compute_adjacency_matrix(surface)
+    return [find(row)[1] for row in matrix]
+        
+def smooth_surface_data(surface, surf_data,
```

Comment:
I thought about that. But I think that this is where the analogy between surface and volume breaks down. smooth_surf and smooth_surface might also refer to smoothing a surface mesh (otherwise referred to as inflating the mesh). So I went with smooth_surface_data with the analogy to load_surface_data. I separately might write a smooth_surface_mesh to smooth the mesh